atc/db/resource_type.go

```diff
@@ -87,11 +90,24 @@ func (resourceTypes ResourceTypes) Deserialize() atc.VersionedResourceTypes {
 	var versionedResourceTypes atc.VersionedResourceTypes
 
 	for _, t := range resourceTypes {
+		// Apply source defaults to resource types
+		source := t.Source()
+		parentType, found := resourceTypes.Parent(t)
+		if found {
+			source = parentType.Defaults().Merge(source)
+		} else {
+			parentType, found, _ := t.FindParentBaseResourceType()
```

Comment:
This `err` should not be ignored - an error reaching the database shouldn't mean that defaults are skipped

Reply:
With the second commit, this is no longer an issue.